cpp/src/arrow/dataset/file_parquet.cc

```diff
@@ -366,8 +366,13 @@ std::optional<compute::Expression> ParquetFileFragment::EvaluateStatisticsAsExpr
     const parquet::Statistics& statistics) {
   auto field_expr = compute::field_ref(field_ref);
 
+  bool may_have_null = !statistics.HasNullCount() || statistics.null_count() > 0;
   // Optimize for corner case where all values are nulls
-  if (statistics.num_values() == 0 && statistics.null_count() > 0) {
+  if (statistics.num_values() == 0) {
+    // If `statistics.HasNullCount()`, it means the all the values are nulls.
+    //
+    // If there are no values and no nulls, it might be empty or all values
+    // are nulls. In this case, we also return a null expression.
```

Comment:
Can you update this comment?

Reply:
Oh, it might be mistake

Reply:
```
    // If there are no values and `!statistics.HasNullCount()`, it might be
    // empty or all values are nulls. In this case, we also return a null
    // expression.
```

Change to this